src/ORM/Connect/MySQLSchemaManager.php

```diff
@@ -647,4 +682,143 @@ protected function defaultClause($values)
+        preg_match('/(STORED|VIRTUAL)$/i', $spec, $matches);
+        return strtoupper($matches[1] ?? '');
+    }
+
+    /**
+     * Normalises a MySQL generated column expression.
+     * This is needed to robustly validate whether the expression we want matches the expression
+     * that's already in the database, because MySQL sometimes makes changes internally that we
+     * need to normalise out, such as:
+     * - escaping quotes
+     * - double-escaping backslashes
+     * - adding the charset before explicit strings
+     * - changing case of keywords
+     * - using backticks instead of ANSI quotes around columns names
+     * - adding spaces around operators
+     */
+    private function normaliseGeneratedColumnExpression(string $expression): string
+    {
+        // Clean up string literals. Easier to do here than in the AST because the
+        // escaped quotes MySQL gives us change the representation of the parsed string.
+        $expression = preg_replace_callback(
+            // Both \' and '' are valid escape sequences for a single quote in a string literal
+            // see https://dev.mysql.com/doc/refman/8.4/en/string-literals.html
+            "/\\\\'(?:[^']|''|\\\\\\\\')*\\\\'/",
+            function ($matches) {
+                // Replace escaped quotes around the string literal with single unescaped quotes
+                $x = preg_replace("/^\\\\'(.*)\\\\'$/", "'$1'", $matches[0]);
+                // Tidy up escaped backslashes, e.g. for FQCN
+                return str_replace('\\\\\\\\', '\\\\', $x);
+            },
+            $expression
+        );
+
+        // Parse and normalise the AST (needs to be a valid SQL command so prepend with SELECT)
+        $parser = new PHPSQLParser(options: [Options::ANSI_QUOTES => true]);
+        $parsed = $parser->parse('SELECT ' . $expression);
+        $this->normaliseExpressionAST($parsed);
+        // Render back out as an SQL expression string
+        $sqlCreator = new PHPSQLCreator($parsed);
+        return preg_replace('/^SELECT /', '', $sqlCreator->created);
+    }
+
+    private function normaliseExpressionAST(array &$ast, ?array &$parent = null, string|int $i = 'root'): void
+    {
+        if (isset($ast['base_expr']) && isset($ast['expr_type'])) {
+            // Normalise casing for reserved keywords
+            if (PHPSQLParserConstants::getInstance()->isReserved(strtoupper($ast['base_expr']))) {
+                $ast['base_expr'] = strtoupper($ast['base_expr']);
+            }
+
+            // Normalise column references
+            if ($ast['expr_type'] === ExpressionType::COLREF) {
+                // Remove unnecessary charset references
+                $charset = '_' . MySQLDatabase::config()->get('charset');
+                if ($ast['base_expr'] === $charset) {
+                    unset($parent[$i]);
+                // Normalise column references to be ANSI quotes
```

Comment:
MySQL will add the default charset before string literals, e.g. if we give it `'my string'` it will transform it into `_utf8mb4\'my string\'`

We want to remove the default charset - but importantly _not_ touch charsets that are not the default, because that will have been added intentionally by a developer.